package/spec/frontend/media_player/hooks_spec.js

```diff
@@ -104,18 +106,23 @@ describe('hooks', function() {
         expect(player.originalPlay).toHaveBeenCalled();
       });
 
-      it('does not call original play method if player is paused before promise returned by before is resolved', function() {
+      it('does not call original play method if player is paused before promise returned by before is resolved', async () => {
         var player = fakePlayer();
-        var deferred = new jQuery.Deferred();
+
+        var promiseResolve;
+        var promise = new Promise(function(resolve, reject) {
+          promiseResolve = resolve;
+        });
+
         hooks(player, {
           before: function() {
-            return deferred.promise();
+            return promise;
           }
         });
 
         player.play();
         player.pause();
-        deferred.resolve();
+        promiseResolve();
 
         expect(player.originalPlay).not.toHaveBeenCalled();
```

Comment:
I think this test also just passes since there is no await. Maybe you did this and I'm missing something, but generally it's often a good idea to try to change some code back to see a changed test fail and make sure it fails for the right reason.